frontend/mainwindow.cpp

```diff
@@ -105,24 +142,89 @@ void MainWindow::on_filterBox_textChanged(const QString &arg1) {
   proxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
 }
 
+// Filter table data
+void MainWindow::on_searchBox_textChanged(const QString &arg1) {
+  std::get<0>(matchesTuple).clear();
+  std::get<1>(matchesTuple) = 0;
+  if (arg1 == "") {
+    ui->runDataTable->selectionModel()->clearSelection();
+    return;
+  }
+  for (int i = 0; i < proxyModel->rowCount(); i++) {
+    if (ui->runDataTable->isColumnHidden(i) == false) {
+      std::get<0>(matchesTuple)
+          .append(proxyModel->match(proxyModel->index(0, i), Qt::DisplayRole,
+                                    arg1, -1, Qt::MatchContains));
+    }
+  }
+  if (std::get<0>(matchesTuple).size() > 0) {
+    ui->runDataTable->selectionModel()->clearSelection();
+    ui->runDataTable->selectionModel()->setCurrentIndex(
+        std::get<0>(matchesTuple)[0],
+        QItemSelectionModel::Select | QItemSelectionModel::Rows);
+  }
+}
+
+void MainWindow::on_findUp_clicked() {
+  if (std::get<0>(matchesTuple).size() > 0) {
+    if (std::get<1>(matchesTuple) >= 1) {
+      std::get<1>(matchesTuple) -= 1;
+    } else {
+      std::get<1>(matchesTuple) = 0;
+    }
+    ui->runDataTable->selectionModel()->clearSelection();
+    ui->runDataTable->selectionModel()->setCurrentIndex(
+        std::get<0>(matchesTuple)[std::get<1>(matchesTuple)],
+        QItemSelectionModel::Select | QItemSelectionModel::Rows);
+  }
+}
+
+void MainWindow::on_findDown_clicked() {
+  if (std::get<0>(matchesTuple).size() > 0) {
+    if (std::get<1>(matchesTuple) < std::get<0>(matchesTuple).size() - 1) {
+      std::get<1>(matchesTuple) += 1;
+    }
+    ui->runDataTable->selectionModel()->clearSelection();
+    ui->runDataTable->selectionModel()->setCurrentIndex(
+        std::get<0>(matchesTuple)[std::get<1>(matchesTuple)],
+        QItemSelectionModel::Select | QItemSelectionModel::Rows);
+  }
+}
+
+void MainWindow::on_searchAll_clicked() {
+  if (std::get<0>(matchesTuple).size() > 0) {
+    ui->runDataTable->selectionModel()->clearSelection();
+    std::get<1>(matchesTuple) = -1;
+    for (int i = 0; i < std::get<0>(matchesTuple).size(); i++) {
+      ui->runDataTable->selectionModel()->setCurrentIndex(
+          std::get<0>(matchesTuple)[i],
+          QItemSelectionModel::Select | QItemSelectionModel::Rows);
+    }
+  }
+}
```

Comment:
Lots of code duplication here. You can create a simple function `goToCurrentFoundIndex()` (or similar) which clears, highlights, and moves to the correct QModelIndex, and call that from the various functions here once you've modified the index. Also on my machine the row isn't visibly selected until you scroll the list manually.  That may be a quirk of Qt, or we're missing a crucial (but not entirely obvious) function call somewhere.

Reply:
The issue with selection is that when the table isn't in focus, on windows, selected items become grey and not blue. It's just a really similar grey to the current row backgrounds so it's hard to see. This is a visibility issue rather than a functional one